### PR TITLE
refactor: move serve state logic into state package

### PR DESCRIPTION
**Pull request by aku11i**
## Summary
- Add a new @phantompane/state package for serve state schemas, types, storage, and tests.
- Move serve state persistence from app server storage into the state package.
- Change the default state path from phantom/serve/state.json to phantom/state.json, without migration or compatibility handling.

## Validation
- pnpm --filter @phantompane/state typecheck
- pnpm --filter @phantompane/state test
- pnpm --filter app-private typecheck
- pnpm --filter app-private test
- pnpm typecheck
- pnpm test
- pnpm lint

## Notes
- pnpm ready was run, but @phantompane/e2e failed because this worktree does not have a phantom executable on PATH; the same 14 e2e cases exit with code 1 before exercising the changed state logic.

### CODE REVIEW

File: packages/state/src/store.ts [hunk 1]
```diff
@@ -31,7 +27,7 @@ export function getDefaultServeDataDir(env = process.env): string {
     stateHome && isAbsolute(stateHome)
       ? stateHome
       : join(homedir(), ".local", "state");
-  return join(baseStateDir, "phantom", "serve");
+  return join(baseStateDir, "phantom");
```
[the remark this answers is not in the document]
Comment (aku11i) [reply packages/state/src/store.ts:30]: This is intentional for this refactor. The requested behavior explicitly excludes migration and backward compatibility support for the old `phantom/serve/state.json` location, so the default path change to `phantom/state.json` should not add a legacy fallback in this PR.

File: packages/state/src/store.ts [hunk 2]
```diff
@@ -142,21 +69,23 @@ export class ServeStateStore {
...
       if (
         error &&
         typeof error === "object" &&
         "code" in error &&
         error.code === "ENOENT"
       ) {
-        this.state = createEmptyState();
+        const state = createEmptyState();
+        await this.save(state);
+        return state;
```
[the remark this answers is not in the document]
Comment (aku11i) [reply packages/state/src/store.ts:84]: Fixed in `f34bfe8`. `load()` now serializes missing-state creation through the same `updateChain` used by `update()`, while `update()` calls a private loader inside the chain to avoid self-deadlock. I also added a regression test that blocks the initial empty-state save and verifies a concurrent update waits and persists its project afterward.
